Sources/Sentry/Public/SentryScope.h

```diff
@@ -113,6 +121,23 @@ NS_SWIFT_NAME(Scope)
  */
 - (void)addAttachment:(SentryAttachment *)attachment;
 
+/**
+ * Sets the scope span with given span only if the scope has no span already.
+ */
+- (void)setSpanIfEmpty:(id<SentrySpan>)span;
+
+/**
+ * Removes the span from scope atomically.
+ */
+- (void)clearSpan;
+
+/**
+ * Removes the span from scope atomically if it is the given span.
+ *
+ * @param span The span that should be removed from scope.
+ */
+- (void)clearSpan:(id<SentrySpan>)span;
```

Comment:
on Java we have a single method that allows you to do whatever you want atomically eg:

`public void withTransaction(final @NotNull IWithTransaction callback)`

it looks much nicer than all this exposed APIs, clearSpan, clearSpan(span), setSpanIfEmpty etc

Reply:
Interesting, It is a good idea to resolve this problem.